sw/device/bsp/w25q/w25q.c

```diff
@@ -838,23 +847,30 @@ w25q_error_codes_t w25q128jw_read_quad_dma(uint32_t addr, void *data, uint32_t l
     // Take into account the extra bytes (if any)
     if (length % 4 != 0) {
         uint32_t last_word = 0;
-        spi_read_word(spi, &last_word);
+      
+        spi_read_word((spi_host_t *)&spi, &last_word);
```

Comment:
could these casts being the issue of the failing spi_write example? @JuanSapriza

Reply:
Not the casting, but the & could. `spi` is already a pointer. Will check now

Reply:
Indeed, it's working now. Will PR to @beeeeep